zio-json/shared/src/test/scala/zio/json/internal/async/DecodingSpec.scala

```diff
@@ -0,0 +1,91 @@
+package zio.json.internal.async
+
+import zio.test._
+
+object DecodingSpec extends DefaultRunnableSpec {
+  val spec: ZSpec[Environment, Failure] =
+    suite("AsyncDecoding")(
+      suite("boolean")(
+        suite("true")(
+          test("no leftovers") {
+            val registers = Registers(Array(()))
+            val decoding  = AsyncDecoder.boolean.unsafeNewDecoding(0, registers)
+
+            val leftovers = decoding.feed("true")
+
+            assertTrue(leftovers == 0) &&
+            assertTrue(registers.registers(0) == true)
+          },
+          test("needs more") {
+            val registers = Registers(Array(()))
+            val decoding  = AsyncDecoder.boolean.unsafeNewDecoding(0, registers)
+
+            val leftovers = decoding.feed("tr")
+
+            assertTrue(leftovers == 2) &&
+            assertTrue(registers.registers(0) == ())
+
+            val oneMore = decoding.feed("u")
+
+            assertTrue(oneMore == 1) &&
+            assertTrue(registers.registers(0) == ())
```

Comment:
Those are unused, or does `assertTrue` raise an exception in ZIO 2 when used with a sync test?

Reply:
It will not raise it, forgot that is is not used 🤦